File: src/multimodal_generator.ts
export type InteractionMode = 'FOCUS' | 'CREATIVE' | 'EMPATHIC';

export interface PromptAnalysis {
    mode: InteractionMode;
    keywordsDetected: string[];
    isTechnical: boolean;
}

export interface MediaGenerationRequest {
    type: 'IMAGE' | 'AUDIO' | 'VIDEO';
    prompt: string;
    contextMode: InteractionMode;
}

export class MultimodalGenerator {
    /**
     * Valida que la generación de contenido multimedia respete el tono de la conversación.
     */
    public validateMediaTone(request: MediaGenerationRequest): string {
        if (request.contextMode === 'FOCUS') {
            return `Modificando prompt para [${request.type}]: Asegurando un estilo sobrio, profesional, minimalista y libre de distracciones o tonos festivos. Base prompt: "${request.prompt}"`;
        }

        if (request.contextMode === 'CREATIVE') {
            return `Modificando prompt para [${request.type}]: Añadiendo energía, creatividad, colores vibrantes y un tono inspirador. Base prompt: "${request.prompt}"`;
        }

        // EMPATHIC
        return `Modificando prompt para [${request.type}]: Usando un tono relajante, equilibrado, amable y acogedor. Base prompt: "${request.prompt}"`;
    }

    public async generateMedia(request: MediaGenerationRequest): Promise<string> {
        const adjustedPrompt = this.validateMediaTone(request);
        // Simulando la llamada al motor de IA generativa (Stable Diffusion, MusicGen, etc.)
        console.log(`[Generando ${request.type}] -> ${adjustedPrompt}`);
        return `Archivo ${request.type.toLowerCase()} generado exitosamente basado en el tono ${request.contextMode}.`;
    }
}

File: src/multimodal_generator.py
class MultimodalGenerator:
    def __init__(self):
        pass

    def validate_media_tone(self, media_type: str, prompt: str, context_mode: str) -> str:
        """
        Garantiza que el tipo de medio (audio/imagen/video) y el prompt sigan el tono de la conversación.
        """
        if context_mode == "FOCUS":
            # Para temas serios o técnicos, bloqueamos el modo festivo
            return f"Modificando prompt para [{media_type}]: Tono sobrio, instrumental de concentración/lofi sin distracciones, o diseño gráfico técnico y limpio. Base: '{prompt}'"
            
        elif context_mode == "CREATIVE":
            # Liberamos creatividad
            return f"Modificando prompt para [{media_type}]: Estilo inspirador, enérgico, colorido o expansivo. Base: '{prompt}'"
            
        else: # EMPATHIC
            # Tono de apoyo y relajante
            return f"Modificando prompt para [{media_type}]: Tono relajante, suave, armonioso e introspectivo. Base: '{prompt}'"

    def generate_media(self, media_type: str, prompt: str, context_mode: str) -> str:
        adjusted_prompt = self.validate_media_tone(media_type, prompt, context_mode)
        
        # Simulación de llamada a la IA de generación de contenido (imágenes, música, etc.)
        print(f"[Simulando API Generativa] \n--> Modo: {context_mode}\n--> Tipo: {media_type}\n--> Prompt calibrado: {adjusted_prompt}")
        
        return f"Generación completada para tipo {media_type} bajo el esquema de modo {context_mode}."
